src/librustc_mir/borrow_check.rs

```diff
@@ -1413,22 +1375,22 @@ impl<'cx, 'gcx, 'tcx> MirBorrowckCtxt<'cx, 'gcx, 'tcx> {
                             return None;
                         };
 
-                        self.tcx
-                            .with_freevars(node_id, |freevars| {
-                                for (v, lv) in freevars.iter().zip(lvs) {
-                                    if let Operand::Consume(Lvalue::Local(l)) = *lv {
-                                        if local == l {
-                                            debug!(
-                                                "find_closure_span: found captured local {:?}",
-                                                l
-                                            );
-                                            return Some(v.span);
-                                        }
+                        self.tcx.with_freevars(node_id, |freevars| {
+                            for (v, lv) in freevars.iter().zip(lvs) {
+                                match *lv {
+                                    Operand::Copy(Lvalue::Local(l)) |
+                                    Operand::Move(Lvalue::Local(l)) if local == l => {
+                                        debug!(
+                                            "find_closure_span: found captured local {:?}",
+                                            l
+                                        );
+                                        return Some(v.span);
                                     }
+                                    _ => {}
```

Comment:
Personally, I'd rather this exhaustive (and perhaps move the `if` on the arm above inside the arm. But I rate the probability of overlooking this `match` after some future change to `Operand` as fairly low, so it's not that important.